datafusion/physical-expr/src/functions.rs

```diff
@@ -436,7 +437,7 @@ pub fn create_physical_fun(
             DataType::LargeUtf8 => {
                 make_scalar_function_inner(string_expressions::ascii::<i64>)(args)
             }
-            other => internal_err!("Unsupported data type {other:?} for function ascii"),
+            other => exec_err!("Unsupported data type {other:?} for function ascii"),
```

Comment:
I'm thinking if its good to use `not_impl_err!` for unsupported errors

Reply:
I could do that. It's unfortunate that this type of error can't be validated and caught at compile time.